src/accountsdb/db.zig

```diff
@@ -649,10 +645,15 @@ pub const AccountsDB = struct {
             }
         }
 
+        // free extra memory, if we overallocated (very likely)
+        if (n_accounts_total != references_buf.len) {
```

Comment:
Should this check not be `n_accounts_total < references_buf.len`? it looks like  if it is the case of underallocation (not sure how likely this would be) an attempt to free extra memory would be made?

Reply:
its kind of equivalent because `n_accounts_total` would never be greater than `references_buf.len`, otherwise another error would have been triggered earlier -- so checking != is the same as <